### PR TITLE
Feat : play.php output file or database deletion errors to user

### PR DESCRIPTION
**Pull request by alexbelgium**
When : 
   - In the recording tab, the user tries to delete an observation
   - the caddy user doesn't have the right access to the sqlite database

What happens : the file gets deleted, but not the sqlite database line, messing up statistics

Why it happens : 
   - The php code silently fails to open the database read_write, which fails silently to delete the line
   - Given that the file does not exist anymore, it doesn't appear anymore in "Recordings" which prevents an easy access for line deletion

Proposal : 
- first check that the sqlite line is deleted before deleting file
- provides user feedback regarding which issue occured

### CODE REVIEW

**Comment by Nachtzuster** [comment]
Not sure flipping the order around is the correct thing to do: if file deletion fails but db deletion succeeds, you end up with orphaned files. Also since the DB is THE reference, it should be done last

The recordings page should probably be fixed to show entries even if the files are missing.

Improving the error reporting is welcome though!

**Comment by alexbelgium** [comment]
Thanks ; good idea ! Actually I had done it in this order because I encountered a bug where the "caddy" user (that executes the php script) did not had the right permissions to the birds.db (owned by "pi"). Therefore, the file was deleted, but not the sqlite entry. This meant that the specie was still appearing in the webui, but that it couldn't be deleted anymore as the file was not found...

And as the script failed without error it took me some time to understood what happened.

Your proposal to fix recordings page is probably the right one, but wouldn't this get messy when the "purge" of old recordings occur due to disk space limitations ? I'll think about it :-)

A shorter term solution could be to check for both sqlite & file permissions before deleting, what do you think ?

**Comment by Nachtzuster** [comment]
> wouldn't this get messy when the "purge" of old recordings occur due to disk space limitations ?

yeah, I thought of this afterward too

Since we're talking about an edgecase here (shouldn't actually happen 'in the field') , I would keep it simple and just make the error reporting better

**Comment by alexbelgium** [comment]
Thanks I've adapted the pr in this direction

**Comment by alexbelgium** [comment]
Seems I closed this by error

**Review by Nachtzuster — CHANGES_REQUESTED** [review]

File: scripts/play.php [hunk 1]
```diff
@@ -28,9 +28,12 @@
   if (!exec("sudo rm $file_pointer && sudo rm $file_pointer.png")) {
```
[the remark this answers is not in the document]
Comment (Nachtzuster) [reply scripts/play.php:28]: @alexbelgium Sorry about the late response, I seem to always forget too _submit_ the review

[the remark this answers is not in the document]
Comment (alexbelgium) [reply scripts/play.php:28]: Thanks i'll redo some tests then

**Comment by alexbelgium** [comment]
Hi, should be corrected now !

File: scripts/play.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexbelgium) [reply scripts/play.php]: Should be good now

**Comment by alexbelgium** [comment]
Tested to be working with only an issue : there should be a line return between the error output of shell script and the error output of sqlite (this is for a non-existant line)

Error - file deletion failed : rm: cannot remove '/home/pi/BirdSongs/Extracted/By_Date/2024-04-29/Pigeon_colombin/Pigeon_colombin-73-2024-04-29-birdnet-RTSP_1-11:38:09.mp3': No such file or directoryError - database line deletion failed : not an error

**Pull request merged**